File: Searching/Sahil/JAVA/LinearSearch.java
public class LinearSearch {

    public static int linearSearch(int[] arr, int numberToSearch) {

        int position = -1 ;
        for (int i = 0; i < arr.length; i++) {
            if(arr[i] == numberToSearch) {
                position = i ;
            }
        }
        return position ;
    }

    public static void main(String[] args) {

        int[] arr = {10, 20, 80, 30, 60, 50,  110, 100, 130, 170} ;
        int numberToSearch = 110 ;
        int searchResult = linearSearch(arr, numberToSearch) ;
        if(searchResult >= 0) {
            System.out.println("Number found at: " + (searchResult + 1));
        } else {
            System.out.println("Number not found");
        }
    }
}

File: Searching/Sahil/C++/linear_search.cpp
#include <iostream>
#include <array>
using namespace std;

int liniar_search(int arr[], int size, int number_to_search) {

    int location = -1 ;
    for (int i = 0; i < size; i++) {
        if(arr[i] == number_to_search) {
            location = i ;
            break ;
        }
    }
    return location ;
}
int main() {

    array<int, 10> arr = {10, 20, 80, 30, 60, 50,  110, 100, 130, 170} ;
    int number_to_search = 110 ;
    cout << "The location of " << number_to_search << " is " << liniar_search(arr.data(), arr.size(), number_to_search) + 1 ;
    return 0;
}
